File: training/jvm/src/main/java/org/corgium/training/Palindrome.java
package org.corgium.training;

public class Palindrome {
    
    // func isPalindrome(x int) bool {
// 	number := strconv.Itoa(x)
// 	j := len(number)
// 	correction := j % 2
// 	for i := range j {
// 		if i == j-i-correction {
// 			return true
// 		} else if number[i] != number[j-i-1] {
// 			return false
// 		}
// 	}
// 	return false
// }
    public boolean isPalindrome(int x) {
        String number = Integer.toString(x);
        int j = number.length();
        int correction = j % 2;
        for (int i = 0; i < j; i++) {
            if (i == j - i - correction) {
                return true;
            } else if (number.charAt(i) != number.charAt(j - i - 1)) {
                return false;
            }
        }
        return false;    
    }
}

File: training/golang/pkg/leetcode/palindrome.go
// isPalindrome checks if the given integer is a palindrome.
package leetcode

import "strconv"

func IsPalindrome(x int) bool {
	number := strconv.Itoa(x)
	j := len(number)
	correction := j % 2
	for i := range j {
		if i == j-i-correction {
			return true
		} else if number[i] != number[j-i-1] {
			return false
		}
	}
	return false
}
